apigee-cf-service-broker/api/service_instances.js

```diff
@@ -120,6 +120,38 @@ function actionValidate(req, res, next) {
   }
 }
 
+function deriveProtocol(params){
+  const proto = params.protocol ? params.protocol.toString().trim().toLowerCase() : ''
+  const ret = {
+    any: false,
+    errors: []
+  }
+  if (proto === "http" || proto === "https"){
+    ret["protocol"] = proto
+    ret.any = true
+  }
+  else if (proto){
+    ret.errors.push(proto)
+  }
+  return ret
+}
+
+function protocolValidate(req, res, next){
+  const proto = deriveProtocol(req.body.parameters)
+  if (proto.errors.length) {
+    res.status(400)
+    var loggerError = logger.ERR_UNSUPPORTED_TARGET_PROTOCOL(null, null, proto.errors.map(quoteAction).join(', '))
+    res.json(loggerError)
+  }
+  else if (proto.any) {
+    next()
+  }
+  else {
+    res.status(400)
+    var loggerError = logger.ERR_MISSING_TARGET_PROTOCOL(null, null, "http or https")
```

Comment:
Options -- valid or invalid -- are quoted, so the message should be: `"http" or "https"`